src/web/components/SignInGate/README.md

```diff
@@ -0,0 +1,389 @@
+# Sign In Gate
+
+## Quick Setup Guide
+
+1. Create AB test switch in [guardian/frontend](https://github.com/guardian/frontend/blob/master/common/app/conf/switches/ABTestSwitches.scala), [docs](https://github.com/guardian/frontend/blob/master/docs/03-dev-howtos/01-ab-testing.md#adding-a-switch)
+2. Add test definition to `src/web/experiments/tests` folder, and import test in the `src/web/experiments/ab-tests.ts` file. Variant name in test definition must be unique.
+3. Import test definition in the `tests` array in the `SignInGateSelector.tsx`
+4. If the test needs a design, make it in the `gateDesigns` folder
+5. Set up individual variants in the `gates` folder, exports the `SignInGateComponent` type. Display rules defined here in the `canShow` method. Helpers in `displayRule.ts`.
+6. Import the `SignInGateComponent` in `SignInGateSelector.tsx` by mapping the variant name to the `SignInGateComponent` in the `testVariantToGateMapping` array.
+7. Add it to Storybook by modifying `SignInGate.stories.tsx`
+8. Update Cypress tests in the `cypress/integration/e2e/sign-in-gate.spec.js`, and any unit tests e.g. `displayRule.test.ts`
+
+## Full Guide
+
+### General
+
+For all possible sign in gates, the following rules should apply:
+
+-   All variant names should be unique across all tests
```

Comment:
Do you think this is worth enforcing at a type level? Using an `enum` or `union` type ?

Reply:
That would only be useful to stop spelling mistakes, but not enforce the uniqueness.

It might still be useful though, as you could do something like:

```ts
// in one file
enum SignInGateVariants {
  MainVariant = 'main-variant-1',
  MainControl = 'main-control-1',
  PatientiaVariant = 'patientia-variant-1',
  PatientiaControl = 'patientia-control-1',
}

// in another file
const variant = SignInGateVariants.MainVariant;
```

And then you'll only need to change the variant name in one place to make a change. 🤔